gramps/gen/display/name.py

```diff
@@ -1288,18 +1287,17 @@ def ifNotEmpty(str,p,s):
             ",".join(param),
         )
         try:
-            exec(s) in globals(), locals()
-            return locals()["fn"]
-        except:
+            result = {}
+            exec(s, globals(), result)
+            return result["fn"]
+        except SyntaxError:
```

Comment:
Note, since the string above only defines a function, I can't see how this could fail in a way other than a syntax error.

Reply:
PS, this file could be written in a way that doesn't use `exec` with some `lambda`s, but as it seemed that this was done for speed, I'm not sure if I should change that without having any benchmarks available. So I wonder if there are any benchmarks for this bit of code?